keystoneclient/middleware/auth_token.py

```diff
@@ -967,42 +943,28 @@ def _build_user_headers(self, token_info):
         :raise InvalidUserToken when unable to parse token object
 
         """
-        auth_ref = access.AccessInfo.factory(body=token_info)
-        roles = ','.join(auth_ref.role_names)
+        #auth_ref = access.AccessInfo.factory(body=token_info)
+        #roles = ','.join(auth_ref.role_names)
 
-        if _token_is_v2(token_info) and not auth_ref.project_id:
-            raise InvalidUserToken('Unable to determine tenancy.')
+        #if _token_is_v2(token_info) and not auth_ref.project_id:
+        #    raise InvalidUserToken('Unable to determine tenancy.')
 
         rval = {
             'X-Identity-Status': 'Confirmed',
-            'X-Domain-Id': auth_ref.domain_id,
-            'X-Domain-Name': auth_ref.domain_name,
-            'X-Project-Id': auth_ref.project_id,
-            'X-Project-Name': auth_ref.project_name,
-            'X-Project-Domain-Id': auth_ref.project_domain_id,
-            'X-Project-Domain-Name': auth_ref.project_domain_name,
-            'X-User-Id': auth_ref.user_id,
-            'X-User-Name': auth_ref.username,
-            'X-User-Domain-Id': auth_ref.user_domain_id,
-            'X-User-Domain-Name': auth_ref.user_domain_name,
-            'X-Roles': roles,
-            # Deprecated
-            'X-User': auth_ref.username,
-            'X-Tenant-Id': auth_ref.project_id,
-            'X-Tenant-Name': auth_ref.project_name,
-            'X-Tenant': auth_ref.project_name,
-            'X-Role': roles,
+            'X-Domain-Id': token_info.get('account_id'),
```

Comment:
No this is actually required. Different modules use different headers for creating context. So it would be safe to set these 3 to same value.